File: nosh_now_apis/Models/Merchant.cs
namespace MyApp.Models
{
    public class Merchant : IModel
    {
        public int Id { get; set;}
        public string DisplayName { get; set;}
        public byte[] Avatar { get; set;}
        public string Phone { get; set;}
        public string Email { get; set;}
        public string OpeningTime { get; set;}
        public string ClosingTime { get; set;}
        public string Coordinator { get; set;}
        public bool Status { get; set;}
        public int AccountId { get; set;}  
        public int CategoryId { get; set;}
        public virtual Account Account{ get; set;}
        public virtual Category Category{ get; set;}
        public virtual ICollection<Order> Orders{ get; set;}
        public virtual ICollection<Food> Foods{ get; set;}
    }
}

File: nosh_now_application/lib/data/models/merchant.dart
import 'package:nosh_now_application/data/models/account.dart';
import 'package:nosh_now_application/data/models/category.dart';

class Merchant {
  int merchantId;
  String displayName;
  String avatar;
  String phone;
  String email;
  String openingTime;
  String closingTime;
  String coordinator;
  bool status;
  Account? account;
  FoodCategory? category;
  Merchant(
      {required this.merchantId,
      required this.displayName,
      required this.email,
      required this.phone,
      required this.avatar,
      required this.openingTime,
      required this.closingTime,
      required this.coordinator,
      required this.status,
      this.category,
      this.account});
  factory Merchant.fromJson(Map<dynamic, dynamic> json) {
    return Merchant(
      merchantId: json['id'],
      displayName: json['displayName'],
      email: json['email'],
      phone: json['phone'],
      avatar: json['avatar'],
      openingTime: json['openingTime'],
      closingTime: json['closingTime'],
      coordinator: json['coordinator'],
      status: json['status'],
      account:
          json['account'] != null ? Account.fromJson(json['account']) : null,
          category:
          json['category'] != null ? FoodCategory.fromJson(json['category']) : null,
    );
  }
   Map<String, dynamic> toJson() => {
        'id': merchantId,
        "displayName": displayName,
        'email': email,
        'phone': phone,
        'avatar': avatar,
        'openingTime': openingTime,
        'closingTime': closingTime,
        'coordinator': coordinator,
        'account': account ?? account!.toJson(),
        'category': category ?? category!.toJson()
      };
}
